src/Setup.php

```diff
@@ -71,11 +71,23 @@ public static function init($modelValues = array(), $pluginCallback = NULL, $log
 
     $pluginDir = dirname(__DIR__) . '/plugins';
     $pluginFiles = array();
-    foreach (['*.civi-setup.php', '*/*.civi-setup.php'] as $pattern) {
-      foreach ((array) glob("$pluginDir/$pattern") as $file) {
-        $key = substr($file, strlen($pluginDir) + 1);
-        $key = preg_replace('/\.civi-setup\.php$/', '', $key);
-        $pluginFiles[$key] = $file;
+    $pluginFileDirs = [
+      'blocks',
+      'checkInstalled',
+      'checkRequirements',
+      'common',
+      'createForm',
+      'init',
+      'installFiles',
+      'installDatabase',
+    ];
+    // Don't include uninstall plugin files, as this need to be called on demand and shoudn't be part of Civi installtion setup
```

Comment:
OK, a few things, in no particular order:

(1) Uninstallation is often handled by the same framework as installation. For example, `dpkg` on Debian/Ubuntu; `rpm` on Redhat/Centos; MSI or InstallShield on Windows; ad nauseum -- all of them define the install and uninstall in tandem.

(2) By definition, plugins are only supposed to register listeners for events. In the case of the `plugins/uninstallSchema/*.php` and `plugins/uninstallFiles/*.php`, these plugins only register listeners for the uninstall events. They're not called as part of regular installation.

(3) For *development/testing*, it's useful to have a "force install" use-case which destroys the old material (config-file, DB-tables) and sets them up cleanly. This corresponds to `cv core:install -f`. For this use-case, it would run `uninstallDatabase => uninstallFiles => installFiles => installDatabase`.

(4) Symfony Event has a mechanism for managing order -- you can [set the priority of each listener when registering it](https://symfony.com/doc/current/components/event_dispatcher.html#connecting-listeners). The priority is a number. As a matter of convention, I usually assign the priority based on these constants: `Setup::PRIORITY_START` (2000), `Setup::PRIORITY_PREPARE` (1000), `Setup::PRIORITY_MAIN` (0), `Setup::PRIORITY_LATE` (-1000), `Setup::PRIORITY_END` (-2000). This proposed commit would have the effect of creating a second layer/mechanism for managing order. IMHO, the second mechanism is less-flexible/more-brittle than Symfony's mechanism, and it increases mental load. (When contemplating the order of listeners, then *only* question on one's mind should be the declared priorities.)

(5) If you're experiencing problems with ordering, two possibilities come to mind: 

* (5a) There's a design-flaw in some component that produces a chicken-egg situation. (Ex: We had that problem where Smarty templates called `ts()` which tried to bootstrap Civi before we were ready to boot.)
* (5b) Your dev-loop is missing the `drush dis civicrm -y` step (see [dev loop](https://github.com/civicrm/civicrm-setup/blob/master/docs/getting-started.md#run-the-dev-loop)).

(6) If there's a problem with the ordering of events, could you possibly describe it in terms of the `--debug-event` listing? For example, suppose there's a sequencing problem among the `installDatabase` listeners. You might print the list of listeners and then consider the impact of moving one up or down.

```
$  ~/src/cv/bin/cv core:install --setup-path=$HOME/src/civicrm-setup  --cms-base-url=http://dmaster.l -f -vvv --debug-event

...snip...

[Event] civi.setup.installDatabase
+-------+------------------------------------------------------------------------------------------------------+
| Order | Callable                                                                                             |
+-------+------------------------------------------------------------------------------------------------------+
| #1    | closure(/Users/totten/src/civicrm-setup/plugins/common/LogEvents.civi-setup.php@31)                  |
| #2    | InstallSchemaPlugin->installDatabase()                                                               |
| #3    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/BootstrapCivi.civi-setup.php@13)     |
| #4    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/SetLanguage.civi-setup.php@13)       |
| #5    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/InstallComponents.civi-setup.php@13) |
| #6    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/InstallExtensions.civi-setup.php@13) |
| #7    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/InstallSettings.civi-setup.php@13)   |
| #8    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/FlushBackdrop.civi-setup.php@13)     |
| #9    | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/FlushWordPress.civi-setup.php@13)    |
| #10   | closure(/Users/totten/src/civicrm-setup/plugins/installDatabase/FlushDrupal.civi-setup.php@13)       |
| #11   | closure(/Users/totten/src/civicrm-setup/plugins/common/LogEvents.civi-setup.php@39)                  |
+-------+------------------------------------------------------------------------------------------------------+
```